### PR TITLE
[ExpressionLanguage] Merge docs about ExpresssionLanguage

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

**Comment by javiereguiluz** [comment]
Thanks Fabien for another nice doc merging PR.

File: components/expression_language/syntax.rst [hunk 1]
```diff
@@ -1,320 +0,0 @@
-.. index::
```
Comment (wouterj) [inline components/expression_language/syntax.rst:1]: I think we should keep th is article separate, as this is much more a reference article than a guide

Comment (javiereguiluz) [reply components/expression_language/syntax.rst:1]: I'm not saying that you are wrong ... but in my opinion, having most of the things related to ExpressionLanguage in a single _(long, but manageable)_ article is better for readers.

And with the recent doc redesign, the sticky TOC will make it easier to browse long articles and quickly move between sections.

Comment (wouterj) [reply components/expression_language/syntax.rst:1]: Yes, for the same type of documentation (and with emphasis on the managable).
But no for different types of documentation imho.

https://documentation.divio.com/

**Pull request merged**